### PR TITLE
[BEX-1052] - Create new topic on MSK.

### PR DESCRIPTION
**Pull request by agkountis**

### CODE REVIEW

**Comment by uw-infra** [comment]
Received terraform plan request
```
Cluster: prod-aws
Module: pubsub/kafka-shared-msk-customer-billing
Path: prod-aws/kafka-shared-msk/customer-billing
Commit ID: a2455d7e5de24613f5eb5aa85d6d0fa98c922a60
Requested At: 2024-09-27T10:26:08Z
```
Do not edit this comment. This message will be updated once the plan run is completed.
To manually trigger plan again please post `@terraform-applier plan prod-aws/kafka-shared-msk/customer-billing` as comment.

**Comment by uw-infra** [comment]
Received terraform plan request
```
Cluster: prod-aws
Module: pubsub/kafka-shared-msk-customer-billing
Path: prod-aws/kafka-shared-msk/customer-billing
Commit ID: a2455d7e5de24613f5eb5aa85d6d0fa98c922a60
Requested At: 2024-09-27T10:26:08Z
```
Do not edit this comment. This message will be updated once the plan run is completed.
To manually trigger plan again please post `@terraform-applier plan prod-aws/kafka-shared-msk/customer-billing` as comment.

**Comment by uw-infra** [comment]
Received terraform plan request
```
Cluster: dev-aws
Module: pubsub/kafka-shared-msk-customer-billing
Path: dev-aws/kafka-shared-msk/customer-billing
Commit ID: a2455d7e5de24613f5eb5aa85d6d0fa98c922a60
Requested At: 2024-09-27T10:26:09Z
```
Do not edit this comment. This message will be updated once the plan run is completed.
To manually trigger plan again please post `@terraform-applier plan dev-aws/kafka-shared-msk/customer-billing` as comment.

**Comment by uw-infra** [comment]
Received terraform plan request
```
Cluster: dev-aws
Module: pubsub/kafka-shared-msk-customer-billing
Path: dev-aws/kafka-shared-msk/customer-billing
Commit ID: a2455d7e5de24613f5eb5aa85d6d0fa98c922a60
Requested At: 2024-09-27T10:26:09Z
```
Do not edit this comment. This message will be updated once the plan run is completed.
To manually trigger plan again please post `@terraform-applier plan dev-aws/kafka-shared-msk/customer-billing` as comment.

File: dev-aws/kafka-shared-msk/customer-billing/topics.tf [hunk 1]
```diff
@@ -133,3 +133,19 @@ resource "kafka_topic" "public_fulfilment_events" {
...
   }
 }
+
+# this topic is used during the transition from exstream to bex bill
+# generation and contains AccountReadyToBeFulfilledEvents
+resource "kafka_topic" "transition_bex_fulfilment_request" {
+  name               = "bex.transition.fulfilment_request"
+  replication_factor = 3
+  partitions         = 5
+  config = {
```
Comment (sbuliarca) [inline dev-aws/kafka-shared-msk/customer-billing/topics.tf:143]: This also needs some `retention.bytes` or `retention.ms`, especially since it is dev

**Comment by uw-infra** [comment]
Received terraform plan request
```
Cluster: dev-aws
Module: pubsub/kafka-shared-msk-customer-billing
Path: dev-aws/kafka-shared-msk/customer-billing
Commit ID: 545fe8bc91ea89ef0736b8e3a9426441cd69c470
Requested At: 2024-09-27T10:46:10Z
```
Do not edit this comment. This message will be updated once the plan run is completed.
To manually trigger plan again please post `@terraform-applier plan dev-aws/kafka-shared-msk/customer-billing` as comment.

**Comment by uw-infra** [comment]
Received terraform plan request
```
Cluster: prod-aws
Module: pubsub/kafka-shared-msk-customer-billing
Path: prod-aws/kafka-shared-msk/customer-billing
Commit ID: 545fe8bc91ea89ef0736b8e3a9426441cd69c470
Requested At: 2024-09-27T10:46:10Z
```
Do not edit this comment. This message will be updated once the plan run is completed.
To manually trigger plan again please post `@terraform-applier plan prod-aws/kafka-shared-msk/customer-billing` as comment.

**Comment by uw-infra** [comment]
Received terraform plan request
```
Cluster: dev-aws
Module: pubsub/kafka-shared-msk-customer-billing
Path: dev-aws/kafka-shared-msk/customer-billing
Commit ID: 4a9ab9f204e6d132e098c7bd4e5bd8208265e352
Requested At: 2024-09-27T10:49:45Z
```
Do not edit this comment. This message will be updated once the plan run is completed.
To manually trigger plan again please post `@terraform-applier plan dev-aws/kafka-shared-msk/customer-billing` as comment.

File: dev-aws/kafka-shared-msk/customer-billing/topics.tf (hunk 1 above)
Comment (agkountis) [reply dev-aws/kafka-shared-msk/customer-billing/topics.tf:143]: Done

**Comment by uw-infra** [comment]
Received terraform plan request
```
Cluster: dev-aws
Module: pubsub/kafka-shared-msk-customer-billing
Path: dev-aws/kafka-shared-msk/customer-billing
Commit ID: 912f6b33934f95ee2ec31522f58338bdf92401a9
Requested At: 2024-09-27T10:51:58Z
```
Do not edit this comment. This message will be updated once the plan run is completed.
To manually trigger plan again please post `@terraform-applier plan dev-aws/kafka-shared-msk/customer-billing` as comment.

**Comment by uw-infra** [comment]
Received terraform plan request
```
Cluster: dev-aws
Module: pubsub/kafka-shared-msk-customer-billing
Path: dev-aws/kafka-shared-msk/customer-billing
Commit ID: 0c6af71e8438ee4541fe891b7dfd32058294a7e9
Requested At: 2024-09-27T13:15:07Z
```
Do not edit this comment. This message will be updated once the plan run is completed.
To manually trigger plan again please post `@terraform-applier plan dev-aws/kafka-shared-msk/customer-billing` as comment.

File: dev-aws/kafka-shared-msk/customer-billing/topics.tf [hunk 2]
```diff
@@ -133,3 +133,21 @@ resource "kafka_topic" "public_fulfilment_events" {
...
+    "remote.storage.enable" = "true"
+    # keep data in hot storage for 1 day
+    "local.retention.ms" = "86400000"
+    "max.message.bytes"  = "104857600"
+    "compression.type"   = "zstd"
+    "cleanup.policy"     = "delete"
+    "retention.ms"       = "604800000"
```
[the remark this answers is not in the document]
Comment (agkountis) [reply dev-aws/kafka-shared-msk/customer-billing/topics.tf:151]: Done

**Comment by uw-infra** [comment]
Received terraform plan request
```
Cluster: dev-aws
Module: pubsub/kafka-shared-msk-customer-billing
Path: dev-aws/kafka-shared-msk/customer-billing
Commit ID: 8cbe36830ecc0b4c6e6ec83f9ce832d7b4c7673e
Requested At: 2024-09-27T13:17:22Z
```
Do not edit this comment. This message will be updated once the plan run is completed.
To manually trigger plan again please post `@terraform-applier plan dev-aws/kafka-shared-msk/customer-billing` as comment.

File: prod-aws/kafka-shared-msk/customer-billing/topics.tf [hunk 3]
```diff
@@ -134,3 +134,22 @@ resource "kafka_topic" "public_fulfilment_events" {
...
+  config = {
+    "remote.storage.enable" = "true"
+    # keep data in hot storage for 1 day
+    "local.retention.ms" = "86400000"
+    "compression.type"   = "zstd"
+    "retention.bytes"    = "8053063680"
+    # allow max 100MB for a message
+    "max.message.bytes" = "104857600"
+    "cleanup.policy"    = "delete"
+    "retention.ms"      = "1209600000"
```
Comment (sbuliarca) [inline prod-aws/kafka-shared-msk/customer-billing/topics.tf:153]: Here too, please

**Comment by uw-infra** [comment]
Received terraform plan request
```
Cluster: prod-aws
Module: pubsub/kafka-shared-msk-customer-billing
Path: prod-aws/kafka-shared-msk/customer-billing
Commit ID: 52556550d34faba3ea11ec54617a940d961c5aaf
Requested At: 2024-09-27T13:24:50Z
```
Do not edit this comment. This message will be updated once the plan run is completed.
To manually trigger plan again please post `@terraform-applier plan prod-aws/kafka-shared-msk/customer-billing` as comment.

**Review by sbuliarca — APPROVED** [review]

**Pull request merged**